pkg/bindfilter/bind_filter.go

```diff
@@ -0,0 +1,322 @@
+	BINDFLT_FLAG_MERGED_BIND_MAPPING uint32 = 0x00000002
+	// Use the binding mapping attached to the mapped-in job object (silo) instead of the default global mapping.
+	BINDFLT_FLAG_USE_CURRENT_SILO_MAPPING uint32 = 0x00000004
+	BINDFLT_FLAG_REPARSE_ON_FILES         uint32 = 0x00000008
+	// Skips checks on file/dir creation inside a non-merged, read-only mapping.
+	// Only usable when READ_ONLY_MAPPING is set.
+	BINDFLT_FLAG_SKIP_SHARING_CHECK uint32 = 0x00000010
+	BINDFLT_FLAG_CLOUD_FILES_ECPS   uint32 = 0x00000020
+	// Tells bindflt to fail mapping with STATUS_INVALID_PARAMETER if a mapping produces
+	// multiple targets.
+	BINDFLT_FLAG_NO_MULTIPLE_TARGETS uint32 = 0x00000040
+	// Turns on caching by asserting that the backing store for name mappings is immutable.
+	BINDFLT_FLAG_IMMUTABLE_BACKING              uint32 = 0x00000080
+	BINDFLT_FLAG_PREVENT_CASE_SENSITIVE_BINDING uint32 = 0x00000100
+	// Tells bindflt to fail with STATUS_OBJECT_PATH_NOT_FOUND when a mapping is being added
+	// but its parent paths (ancestors) have not already been added.
+	BINDFLT_FLAG_EMPTY_VIRT_ROOT         uint32 = 0x00000200
+	BINDFLT_FLAG_NO_REPARSE_ON_ROOT      uint32 = 0x10000000
+	BINDFLT_FLAG_BATCHED_REMOVE_MAPPINGS uint32 = 0x20000000
+)
+
+//nolint:revive // var-naming: ALL_CAPS
+const (
+	BINDFLT_GET_MAPPINGS_FLAG_VOLUME uint32 = 0x00000001
+	BINDFLT_GET_MAPPINGS_FLAG_SILO   uint32 = 0x00000002
+	BINDFLT_GET_MAPPINGS_FLAG_USER   uint32 = 0x00000004
+)
+
+// ApplyFileBinding creates a global mount of the source in root, with an optional
+// read only flag.
+// The bind filter allows us to create mounts of directories and volumes. By default it allows
+// us to mount multiple sources inside a single root, acting as an overlay. Files from the
+// second source will superscede the first source that was mounted.
+// This function disables this behavior and sets the BINDFLT_FLAG_NO_MULTIPLE_TARGETS flag
+// on the mount.
+func ApplyFileBinding(root, source string, readOnly bool) error {
+	// The parent directory needs to exist for the bind to work. MkdirAll stats and
+	// returns nil if the directory exists internally so we should be fine to mkdirall
+	// every time.
+	if err := os.MkdirAll(filepath.Dir(root), 0); err != nil {
+		return err
+	}
+
+	if strings.Contains(source, "Volume{") && !strings.HasSuffix(source, "\\") {
+		// Add trailing slash to volumes, otherwise we get an error when binding it to
+		// a folder.
+		source = source + "\\"
+	}
+
+	flags := BINDFLT_FLAG_NO_MULTIPLE_TARGETS
+	if readOnly {
+		flags |= BINDFLT_FLAG_READ_ONLY_MAPPING
+	}
+
+	// Set the job handle to 0 to create a global mount.
+	if err := bfSetupFilter(
+		0,
+		flags,
+		root,
+		source,
+		nil,
+		0,
+	); err != nil {
+		return fmt.Errorf("failed to bind target %q to root %q: %w", source, root, err)
+	}
+	return nil
+}
+
+// RemoveFileBinding removes a mount from the root path.
+func RemoveFileBinding(root string) error {
```

Comment:
Might as well rename ```bfRemoveMapping``` to ```RemoveFileBinding``` and remove this function. What do you think?

Reply:
dealers choice, but if there really isnt any extra setup needed, then that makes the most sense

Reply:
ahh wait. Spoke too soon. The ```bfRemoveMapping()``` function has the job handle param. Let's leave this as is.